ddtrace/internal/dogstatsd.py

```diff
@@ -1,12 +1,15 @@
-from typing import Dict
```

Comment:
These changes seem unrelated to unvendoring six

Reply:
You wish… they are.

Unvendoring six raises a new bunch of mypy errors. In this particular case, there are a lot of typing information that is lost by moving from a URL `str` to a `Dict` where values have multiple types. That's why I removed this parsing function and merged it, so we don't lose typing information.

Reply:
Gotcha! I think it makes sense to do the change separately though

Reply:
Sure, done in https://github.com/DataDog/dd-trace-py/pull/2289